### PR TITLE
eddie-ui: init at 2.20.0

### PR DESCRIPTION
**Pull request by vanbeast**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/73233
Used build scripts for Arch as a reference. 
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:21]:
```suggestion
{ stdenv
, fetchFromGitHub,
,  clang
,  cmake
....
}:
```

I am lazy now but the formatting should be like the above example.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:25]:
```suggestion
  version = "2.18.9";

```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:29]:
```suggestion

  src = fetchFromGitHub {
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:32]:
```suggestion
    rev = version;
```

Otherwise we won't get auto updates.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:35]:
```suggestion

```

IIRC this is not required here.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:38]:
```suggestion
```

Nix auto cds into the extracted directory.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:39]:
```suggestion
    find . -type f -iname "*.sh" -exec chmod +x {} \;
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:40]:
```suggestion
    patchShebangs .
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:43]:
```suggestion
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:44]: This is only available on Darwin.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:47]: NIX_CFLAGS_COMPILE is usually set before the buildPhase.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:49]: This is usually done in postPatch.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:53]: This is usually done in postPatch.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:56]:
```suggestion
  nativeBuildInputs = [ dotnet-sdk pkg-config openvpn mono stunnel ];
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:57]:
```suggestion
  buildInputs = [ which gtk2 lzma.dev glibc libappindicator-gtk2.dev  
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:60]:
```suggestion
    libselinux libsepol fribidi libthai libdatrie xorg.libXdmcp
  ]; 

  installPhase = ''
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:63]:
```suggestion
```
-p creates the directories recursive.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:67]:
```suggestion
    install -Dm755 repository/linux_arch/bundle/eddie-ui/usr/bin/eddie-ui $out/bin/eddie-ui
    substituteInPlace $out/bin/eddie-ui --replace '/usr/' $out/
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:87]:
```suggestion
    changelog = "https://github.com/AirVPN/Eddie/releases";
```

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:89]: gpl3 is deprecated. Please set gpl3Plus or gpl3Only.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (vanbeast) [reply pkgs/applications/networking/eddie-ui/default.nix:67]: Regarding this part. I wanted to ask maybe there is a better way instead of copying Arch script. The script is:
`mono /usr/lib/eddie-ui/eddie-ui.exe --path.resources=/usr/share/eddie-ui --path.exec="$0" "$@"`.
Can we use some Nix wrapper function here?

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (vanbeast) [reply pkgs/applications/networking/eddie-ui/default.nix:35]: If I remove this line, I need to add 
`dontUseCmakeConfigure=true;`
Otherwise I get the error:
`CMake Error: The source directory "/build/source" does not appear to contain CMakeLists.txt.`
Also, when I skip configure phase with the option above I have an error while building:
```
/build/source/src/UI.GTK.Linux.Tray/src/main.cpp:22:10: fatal error: libappindicator/app-indicator.h: No such file or directory
   22 | #include <libappindicator/app-indicator.h>
```
Looks like in this case I need to provide some cmake flags manually.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (vanbeast) [reply pkgs/applications/networking/eddie-ui/default.nix:44]: Sorry, didn't get it. Could you please explain? Build is fine on NixOS on my Dell laptop.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please fix the merge conflict and do not merge master into PRs.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:28678]: Please sort this properly.

**Comment by vanbeast** [comment]
Hi @SuperSandro2000 
Sorry, didn't have time to fix the PR. Now it's updated. Could you please check buildInputs and nativeBuildInputs? It needs mono, openvpn and stunnel at runtime. Is it correct to put them in buildInputs?

**Comment by vanbeast** [comment]
@SuperSandro2000 Did some cleanup according to your comments. Please review

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/networking/eddie-ui/default.nix:79]: I am not sure but it could be that the binary gets double wrapped.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/eddie-ui/default.nix:98]:
```suggestion
    longDescription = ''
        Eddie is a OpenVPN GUI for AirVPN. AirVPN is based on OpenVPN and operated by activists and hacktivists in defence of net neutrality, privacy and against censorship.
    '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix
Comment (vanbeast) [reply maintainers/maintainer-list.nix]: Do you mean separate PR or just separate commit in this PR?

**Comment by vanbeast** [comment]
@SuperSandro2000 
`    substituteInPlace src/App.CLI.Linux.Elevated/build.sh --replace '-static -pthread' '-pthread'
`
Regarding the line above. I'm removing -static option because otherwise I have these errors:
...binutils-2.35.1/bin/ld: cannot find -lpthread
...binutils-2.35.1/bin/ld: cannot find -lm
...binutils-2.35.1/bin/ld: cannot find -lpthread
...binutils-2.35.1/bin/ld: cannot find -lc

Do you have an idea how it can be fixed? 
Also, we want packages mono, openvpn and stunnel installed together with eddie-ui. Can we add something to default.nix to ensure this?

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000
> ` substituteInPlace src/App.CLI.Linux.Elevated/build.sh --replace '-static -pthread' '-pthread'`
> Regarding the line above. I'm removing -static option because otherwise I have these errors:
> ...binutils-2.35.1/bin/ld: cannot find -lpthread
> ...binutils-2.35.1/bin/ld: cannot find -lm
> ...binutils-2.35.1/bin/ld: cannot find -lpthread
> ...binutils-2.35.1/bin/ld: cannot find -lc
> 
> Do you have an idea how it can be fixed?
> Also, we want packages mono, openvpn and stunnel installed together with eddie-ui. Can we add something to default.nix to ensure this?

I don't know.

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:10973]: Please put this into a separate commit with the message ``maintainers: add vanbeast `` and sort it alphabetically.

File: pkgs/applications/networking/eddie-ui/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/networking/eddie-ui/default.nix:78]: makeWrapper has an ``--add-flags`` option which can do that.

**Comment by vanbeast** [comment]
Thanks Sandro. I made this PR a draft. I will need to spend some time to figure out those issues

**Comment by chrisdebian** [comment]
Hi, guys.

Is the work in AirVPN/ eddie-ui dead? I'm considering distrohopping and making NixOS my main OS. I do need AirVPN/ eddie, though. I guess it's probably available as a Flatpak, if not from NixOS.

Thanks,

Chris

**Comment by DavidOliver** [comment]
I'd also like to use Eddie. In the meantime, I've found that `sudo openvpn <openvpn config file downloaded from airvpn's config generator tool>` works well. I believe firewall rules can block traffic other than that of the tun.

**Comment by vanbeast** [comment]
I've stopped using NixOS, so I'm not going to work on this package anymore

**Pull request closed without merging**
